migrate/scripts/object-id.js

```diff
@@ -0,0 +1,42 @@
+var mongodb = require("mongodb"),
+    ObjectId = mongodb.ObjectId
+
+const canRun = collection => (
+  new Promise((resolve, reject) => {
+    collection.find({
+      '_id': {
+        $type: "string"
+      }
+    })
+    .toArray()
+    .then(r => {
+      const ids = r.map(e => e._id);
+
+      resolve(ids.some(i => i));
+    });
+  })
+);
+
+const run = collection => {
+  let itr = 0;
+  collection.find({
+    '_id': {
+      $exists: true
+    }
+  })
+  .toArray()
+  .then(idEntries => {
+    idEntries.forEach(idEntry => {
+      if (typeof idEntry._id === "string") {
+        let newId = ObjectId(idEntry._id);
+        let oldId = idEntry._id;
+        idEntry._id = newId;
+        collection.insertOne(idEntry);
+        collection.deleteOne({ _id: oldId });
```

Comment:
Could we use `update` here?  I think all of my migration scripts use `update`, and I think it will make this script a lot more efficient to only use one database call.

Reply:
I don't think `update` can be used because we're modifying the ID. Based on the research I did the only/proper way to change a documents ID is to create a duplicate of it with the new ID and remove the old one, as I did here.

Reply:
Sounds like it.  Thanks for correcting me there.